source/routing/tests/raptor_test.cpp

```diff
@@ -1495,3 +1495,35 @@ BOOST_AUTO_TEST_CASE(forbid_transfer_between_2_odt){
     auto res1 = raptor.compute(d.stop_areas[0], d.stop_areas[4], 7900, 0, DateTimeUtils::inf, false, true);
     BOOST_REQUIRE_EQUAL(res1.size(), 0);
 }
+
+// 1
+//  \  A
+//   2 --> 3
+//   2 <-- 3
+//  /  B
+// 4
+//
+// We want 1->2->4 and not 1->3->4 (see the 7-pages ppt that explain the bug)
+BOOST_AUTO_TEST_CASE(y_line_the_ultimate_political_blocking_bug){
+    ed::builder b("20120614");
+    b.vj("A")("stop1", 8000, 8000)("stop2", 8100, 8100)("stop3", 8200, 8200);
+    b.vj("B")("stop3", 8400, 8400)("stop2", 8500, 8500)("stop4", 8600, 8600);
+    b.connection("stop1", "stop1", 10);
+    b.connection("stop2", "stop2", 10);
+    b.connection("stop3", "stop3", 10);
+    b.connection("stop4", "stop4", 10);
+    b.data->pt_data->index();
+    b.data->build_raptor();
+    RAPTOR raptor(*(b.data));
+    const type::PT_Data& d = *b.data->pt_data;
+
+    auto res1 = raptor.compute(d.stop_areas[0], d.stop_areas[3], 7900, 0, DateTimeUtils::inf, false, false);
+    BOOST_REQUIRE_EQUAL(res1.size(), 1);
+    BOOST_REQUIRE_EQUAL(res1[0].items.size(), 3);
+    BOOST_CHECK_EQUAL(res1[0].items[0].departure.time_of_day().total_seconds(), 8000);
+    BOOST_CHECK_EQUAL(res1[0].items[0].arrival.time_of_day().total_seconds(), 8100);
+    BOOST_CHECK_EQUAL(res1[0].items[1].departure.time_of_day().total_seconds(), 8100);// A stop2
+    BOOST_CHECK_EQUAL(res1[0].items[1].arrival.time_of_day().total_seconds(), 8500);// B stop2
+    BOOST_CHECK_EQUAL(res1[0].items[2].departure.time_of_day().total_seconds(), 8500);
+    BOOST_CHECK_EQUAL(res1[0].items[2].arrival.time_of_day().total_seconds(), 8600);
```

Comment:
That's funny, in your comment you talk about stop names, and you're checking hours.